lib/src/intl_suggestors/utils.dart

```diff
@@ -0,0 +1,346 @@
+  return s.replaceAll(apos, backslash + apos);
+  // return s.replaceAllMapped(apos, (m) => s[m.start-1] == '\\' ? s[m.start] : backslash + apos);
+}
+
+String removeInterpolationSyntax(String s) =>
+    s.replaceAll(RegExp(r'(\$[^a-zA-Z0-9.]|\s|}|\?.*|\$)'), '');
+
+String toNestedName(String s) =>
+    removeInterpolationSyntax(s)
+        .split('.')
+        .last;
+
+void addMethodToClass(File outputFile, String content) {
+  if (!outputFile.readAsStringSync().contains(content)) {
+    outputFile.writeAsStringSync(content, mode: FileMode.append);
+  }
+}
+
+String toClassName(String str) {
+  final res = toAlphaNumeric(str.replaceAll('_', ' '));
+  final pascalString = res
+      .splitMapJoin(RegExp(r'(?:^\w|[A-Z]|\b\w|\s+)'),
+      onMatch: (m) => m[0]!.toUpperCase())
+      .replaceAll(' ', '');
+  return '${pascalString}Intl';
+}
+
+String toVariableName(String str) {
+  String strippedName = str.replaceFirst(RegExp(r'^[0-9]*'), '').trim();
+  var fiveAtMost = strippedName.split(' ');
+  fiveAtMost = fiveAtMost.sublist(0, fiveAtMost.length < 5 ? fiveAtMost.length : 5);
+  final alphaNumericName = toAlphaNumeric(fiveAtMost.join(' '));
+  final name = toCamelCase(alphaNumericName);
+
+  bool isKeyWord = Keyword.keywords.values
+      .where((keywordValue) => keywordValue.lexeme == name)
+      .isNotEmpty;
+  return isKeyWord ? '${name}String' : name;
+}
+
+String toAlphaNumeric(String str) =>
+    str.replaceAll(RegExp(r'[^a-zA-Z0-9 ]'), '');
+
+String toCamelCase(String str) {
+  /// Remove any non-alphanumeric characters, but leave spaces so we can match by word
+  var res = toAlphaNumeric(str);
+
+  /// BuildCamelCase
+  var capitalizationResult = res.replaceAllMapped(
+      RegExp(r'(?:^\w|[A-Z]|\b\w|\s+)'),
+      (m) =>
+          m.start == 0 ? m[0]?.toLowerCase() ?? '' : m[0]?.toUpperCase() ?? '');
+
+  /// Finally remove the spaces
+  return capitalizationResult.replaceAll(' ', '');
+}
+
+String? getTestId(String? testId, AstNode node) {
+  if (testId != null) return testId;
+  if (node is InvocationExpression) {
+    var component = getComponentUsage(node);
+    if (component != null) {
+      for (final method in component.cascadedMethodInvocations) {
+        if (method.methodName.name == 'addTestId') {
+          final expression = method.node.argumentList.arguments.first;
+          testId = toVariableName(expression
+              .toString()
+              .replaceAll("'", '')
+              .split('.')
+              .last
+              .replaceAll('TestId', ''));
+        }
+      }
+    }
+  }
+
+  if (node.parent != null) {
+    return getTestId(testId, node.parent!);
+  } else {
+    return null;
+  }
+}
+
+bool excludeUnlikelyExpressions<E extends Expression>(PropAssignment prop,
+    String propKey) {
+  if (prop.rightHandSide.staticType == null) return true;
+  if (prop.rightHandSide.staticType?.isDartCoreBool ?? false) return true;
+  if (prop.rightHandSide.staticType?.isDartCoreNull ?? false) return true;
```

Comment:
FYI if you assign to a local variable, the type can be promoted to non-nullable (or subtypes) based on if-checks.
```suggestion
  final staticType = prop.rightHandSide.staticType;
  if (staticType == null) return true;
  if (staticType.isDartCoreBool) return true;
  if (staticType.isDartCoreNull) return true;
```

Reply:
Like really hard